### ISSUE TITLE
.onLoad hardcodes env name

### ISSUE THREAD

**Issue by tschutte** [mlverse/pysparklyr#27]
It appears that the .onLoad function has the python environment name hardcoded, which means that if an environment is specified in either install_psypark or spark_connect, the env name does not appear to used.

It seems that either .onLoad needs to not hardcode the name, or we should not be allowed to specify the name in the other functions to avoid confusion.

**Comment by edgararuiz** [comment]
Hi yes, @t-kalinowski can explain this better. But the best way I can, is to say that the `.onLoad()` call is more of a "if the user doesn't specify anything differently, use `r-sparklyr`", so nothing will be forced to load until we take an action, such as import a library. I just tested it with a different virtual environment to confirm

``` r
# pysparklyr::install_pyspark("test")
# ... restart R
library(sparklyr)
#> 
#> Attaching package: 'sparklyr'
#> The following object is masked from 'package:stats':
#> 
#>     filter
sc <- spark_connect(
  master = "sc://localhost",
  method = "spark_connect",
  envname = "test"
  )
#> ✔ Using the 'test' virtual environment (/Users/edgar/.virtualenvs/test/bin/python)
pysparklyr::installed_components()
#> 
#> ── R packages
#> • `sparklyr` (1.8.3)
#> • `pysparklyr` (0.0.0.9004)
#> 
#> ── Python executable
#> /Users/edgar/.virtualenvs/test/bin/python
#> 
#> ── Python libraries
#> • databricks-connect (13.3.0)
#> • pyspark (3.4.1)
spark_disconnect(sc)
```

<sup>Created on 2023-09-01 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Feel free to try this out, and let me know if it doesn't work for you. 

Thanks!

**Comment by tschutte** [comment]
Hmm, ok.

I initially was having trouble getting things to work with a custom env name until I started over and let it use all of the defaults during the install python step (except for specifying conda).

I'll give it another try, perhaps I had a bad environment due to my prior experiments.

**Comment by t-kalinowski** [comment]
```r
.onLoad <- function(...) {
  use_virtualenv("r-sparklyr", required = FALSE)
  use_condaenv("r-sparklyr", required = FALSE)
}
```
The `required = FALSE` in those calls tells reticulate that this is just a soft preference for that environment name. It just changes the default fallback from being an environment named "r-reticulate" to being an environment named "r-sparklyr".

It's very low on the [order of discovery](https://rstudio.github.io/reticulate/articles/versions.html#order-of-discovery) (at position 9). If you expresses to reticulate a preference for another environment, the `use_virtualenv()` and `use_condaenv()` calls in `.onLoad()` are ignored.

**Comment by tschutte** [comment]
Got it, now I understand what is happening.

So in order to use a custom env name, I have to both call reticulate::use_condaenv("<envname>") in addition to specifying the envname in the call to spark_connect.

Now that I do this, the connection works as expected.

So maybe the issue here is simply a note that the documentation needs to be adjusted to reflect the need to call use_condaenv (or use_virtualenv) in addition to specifying the envname in calls to install_pyspark and spark_connect.

**Comment by edgararuiz** [comment]
Conda environment support is being formalized here: https://github.com/mlverse/pysparklyr/tree/updates

**Comment by edgararuiz** [comment]
Fixed by #35
